PE-1026-20220624: Weekly Executive (Görli)

julienmartinlevrai opened the pull request:
# Description

# Contribution Checklist

- [x] PR title starts with `(PE-<TICKET_NUMBER>)`
- [ ] Code approved
- [ ] Tests approved
- [ ] CI Tests pass

# Checklist

- [ ] Every contract variable/method declared as public/external private/internal
- [ ] Consider if this PR needs the `officeHours` modifier override
- [ ] Verify expiration (`30 days` unless otherwise specified)
- [ ] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [ ] Validate all addresses used are in Goerli changelog or known
- [ ] Notify any external teams affected by the spell so they have the opportunity to review
- [ ] Deploy spell to Goerli `ETH_GAS="XXX" ETH_GAS_PRICE="YYY" make deploy`
- [ ] Ensure contract is verified on `Goerli` etherscan
- [ ] Change test to use Goerli spell address and deploy timestamp
- [ ] Cast spell on Goerli `make spell="0x-deployed-spell-address" cast-spell`
- [ ] Run `make archive-spell` or `make date="YYYY-MM-DD" archive-spell` to make an archive directory and copy `Goerli-DssSpell.sol`, `Goerli-DssSpell.t.sol`, `Goerli-DssSpell.t.base.sol`, and `Goerli-DssSpellCollateralOnboarding.sol`
- [ ] `squash and merge` this PR

talbaneth commented on src/Goerli-DssSpell.sol:
```diff
@@ -23,42 +23,19 @@ pragma solidity 0.6.12;
-import { DssSpellCollateralOnboardingAction } from "./Goerli-DssSpellCollateralOnboarding.sol";
+import { DssSpellCollateralAction } from "./Goerli-DssSpellCollateral.sol";
 
 interface VatLike {
     function Line() external view returns (uint256);
     function file(bytes32, uint256) external;
     function ilks(bytes32) external returns (uint256 Art, uint256 rate, uint256 spot, uint256 line, uint256 dust);
 }
 
```
This can be removed.

julienmartinlevrai replied:
done

talbaneth commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -0,0 +1,192 @@
+    //    https://ipfs.io/ipfs/QmTRiQ3GqjCiRhh1ojzKzgScmSsiwQPLyjhgYSxZASQekj
+    //
+
+    // --- Math ---
+
```
```suggestion
```

talbaneth commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -0,0 +1,192 @@
+        DssExecLib.setKeeperIncentiveFlatRate("UNI-A", 0);
+        DssExecLib.linearInterpolation({
+            _name:      "UNI-A Offboarding",
+            _target:    DssExecLib.spotter(),
```
Neat - I know this was done like this in the previous offboardings, but no need to read from the changelog each time, can store this value locally and use in all the places.

talbaneth reviewed (commented):
Added some neat comments.

talbaneth commented:
@julienmartinlevrai apart from the neat comments and the possible adjustment of end mat (up to risk) this looks good.
Tests pass.

julienmartinlevrai commented:
> @julienmartinlevrai apart from the neat comments and the possible adjustment of end mat (up to risk) this looks good. Tests pass.

addressed your comments @talbaneth 
mat adjustments still pending on my side

godsflaw commented on src/Goerli-DssSpellCollateral.sol:
```diff
@@ -0,0 +1,181 @@
+        // https://forum.makerdao.com/t/signal-request-offboard-uni-univ2daieth-univ2wbtceth-univ2unieth-and-univ2wbtcdai/15160
+
+        DssExecLib.setIlkLiquidationPenalty("UNI-A", 0);
+        DssExecLib.setKeeperIncentiveFlatRate("UNI-A", 0);
```
We set `tip` to `0` as explicitly requested, but `chip` is left unchanged.  From the wording in the offboarding document we make this change to avoid farming liquidation incentives.  While we're unlikely to find ourselves in this farming situation, we may not need to pay for these to offboarding at all.  I think it merits consideration that we set `chip` to `0` too.  This means, like with `TUSD`, we would need to kick these auctions when they were ready.  Note, we did not appear we adjusted `tip` or `chip` when we offboarded other collateral types.  I'm going to just put this here.  I think it's fine if governance pays for the `bark()`, but we probably should have given the `tip` (flat rate) incentive rather than the `chip` (percentage).

godsflaw commented on src/test/starknet.t.sol:
```diff
@@ -0,0 +1,158 @@
+        uint256 dai_bridge_maxDeposit;
+    }
+
+    function setValues() public {
+        starknetValues = StarknetValues({
+            core_implementation:       0x60C5fA1763cC9CB9c7c25458C6cDDFbc8F125256,
+            dai_bridge_isOpen:         1,        // 1 open, 0 closed
+            dai_bridge_ceiling:        200_000,  // Whole Dai Units
+            dai_bridge_maxDeposit:     5         // Whole Dai Units
+        });
+    }
```
I think we should lift this to the `config.sol`

godsflaw replied:
Okay, after some discussion, I think it makes sense to break this out once it becomes large, and even it should be broken out into its own starknet config file.  Once we have MCD on these L2s, the configs should be broken out per L2.

godsflaw reviewed (approved):
LGTM I left a comment on lifting the starknet config values over to the config.sol, but won't block this PR on that.

```haskell
Running 19 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testCastCost() (gas: 2378160)
[PASS] testFailTooEarly() (gas: 399599)
[PASS] testFailTooLate() (gas: 399694)
[PASS] testFailWrongDay() (gas: 399624)
[PASS] testFail_notScheduled() (gas: 14169)
[PASS] testNewChainlogValues() (gas: 2396201)
[PASS] testOnTime() (gas: 2374571)
[PASS] testSpellIsCast_GENERAL() (gas: 24597023)
[PASS] testUNIALerpOffboarding() (gas: 2411536)
[PASS] testUNIV2DAIETHALerpOffboarding() (gas: 2411536)
[PASS] testUNIV2WBTCDAIALerpOffboarding() (gas: 2411471)
[PASS] testUNIV2WBTCETHALerpOffboarding() (gas: 2411557)
[PASS] test_auth() (gas: 14585912)
[PASS] test_auth_in_sources() (gas: 5935637)
[PASS] test_bytecode_matches() (gas: 1692653)
[PASS] test_chainlog_values() (gas: 7116233)
[PASS] test_chainlog_version_bump() (gas: 4091584)
[PASS] test_nextCastTime() (gas: 407376)
[PASS] test_use_eta() (gas: 340464)
Test result: ok. 19 passed; 0 failed; finished in 184.35s
```

godsflaw reviewed (approved):
LGTM can cast after @talbaneth approves

- spell is verified
- spell code matches repo
- DssExecLib matches

```haskell
Running 1 test for src/test/starknet.t.sol:StarknetTests
[PASS] testStarknet() (gas: 2571110)
Test result: ok. 1 passed; 0 failed; finished in 83.15s

Running 19 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testCastCost() (gas: 2378160)
[PASS] testFailTooEarly() (gas: 399599)
[PASS] testFailTooLate() (gas: 399694)
[PASS] testFailWrongDay() (gas: 399624)
[PASS] testFail_notScheduled() (gas: 14169)
[PASS] testNewChainlogValues() (gas: 2396201)
[PASS] testOnTime() (gas: 2374571)
[PASS] testSpellIsCast_GENERAL() (gas: 26274386)
[PASS] testUNIALerpOffboarding() (gas: 2411536)
[PASS] testUNIV2DAIETHALerpOffboarding() (gas: 2411536)
[PASS] testUNIV2WBTCDAIALerpOffboarding() (gas: 2411471)
[PASS] testUNIV2WBTCETHALerpOffboarding() (gas: 2411557)
[PASS] test_auth() (gas: 14585912)
[PASS] test_auth_in_sources() (gas: 5935637)
[PASS] test_bytecode_matches() (gas: 1692653)
[PASS] test_chainlog_values() (gas: 7116233)
[PASS] test_chainlog_version_bump() (gas: 4091584)
[PASS] test_nextCastTime() (gas: 407376)
[PASS] test_use_eta() (gas: 340464)
Test result: ok. 19 passed; 0 failed; finished in 185.41s
```

talbaneth reviewed (commented):
Some minor comments for the starknet test, should be addressed before merging (not blocking spell execution).

Other than that:
* recent changes ok
* deployed spell matches code
* recent mat numbers according to risk calcs
* tests pass

julienmartinlevrai commented:
@talbaneth I added your suggestions

talbaneth commented on src/test/starknet.t.sol:
```diff
@@ -0,0 +1,158 @@
+        checkStarknetCore();
+    }
+
+    function checkStarknetEscrowMom() public {
```
Again sorry for the neatiness :), but you can align the error messages to each other as well.

julienmartinlevrai replied:
no worries :)

godsflaw reviewed (approved):
LGTM ready for merge

talbaneth reviewed (approved):
Good to merge.
